### ISSUE TITLE
Run code generation in parallel

### ISSUE THREAD

**Issue by bigdnf** [AArnott/CodeGeneration.Roslyn#226]
Current implementation of [Codegen](https://github.com/AArnott/CodeGeneration.Roslyn/blob/84f9d1ea9f0cb4ea66534dccadb5997011a64183/src/CodeGeneration.Roslyn.Engine/CompilationGenerator.cs#L103) is using simple loop for code generation. Maybe it would be better to use Parallel.For or other multithread mechanism?

**Comment by amis92** [comment]
@bigdnf Hi.

Parallelism introduces a lot of issues we'd need to resolve. First we'd need to get some hard numbers. How long does it take for generators to run, in comparison with the whole run time of the Tool? Are these costs comparable?

If we can definitely prove that code generation takes most of the time, and not the warm-up and actual compilation before generators are invoked, we can proceed with implementation.

For exactly that purpose, I'll open a new issue to add performance tracking somehow into the Tool. Total run time of the Tool is registered by MSBuild anyway. So then we'll be able to compare these values.
